addons/website/static/src/js/content/menu.js

```diff
@@ -8,6 +8,36 @@ const extraMenuUpdateCallbacks = [];
 const weUtils = require('web_editor.utils');
 const dom = require('web.dom');
 
+publicWidget.registry.NavElementActivator = publicWidget.Widget.extend({
+    selector: "nav",
+
+    /**
+     * @override
+     */
+    start() {
+        for (const anchorEl of this.el.querySelectorAll(
+            "a:not(.dropdown-toggle, .o_mega_menu_toggle, .o_mega_menu_toggle *)")
+        ) {
+            anchorEl.classList.toggle("active", anchorEl.pathname === window.location.pathname);
+        }
+        // TODO: check why the parent is never activated in 16.0 but it is in 17.0.
+        for (const dropdownEl of this.el.querySelectorAll(
+            "a.dropdown-toggle:not(.o_mega_menu_toggle, .o_mega_menu_toggle *)")
+        ) {
+            dropdownEl.classList.toggle("active",
+                dropdownEl.parentNode.querySelector("ul a.active"));
+        }
```

Comment:
I'm not sure we should have this part in the 16.0 part. In the production 16.0, when a dropdown contains an active element, it's not activated. I saw you fixed that in [this other commit](https://github.com/odoo/odoo/commit/8d4e99c6a6e2281b00ec25e870065d0728e27606). But your fix targets 16.2 and not 16.0. I hesitate if I should remove this part from the 16.0 and add it back from 16.2 + forward ports or is it ok to change this small behavior in stable ? What's your opinion @rdeodoo ?

Reply:
Actually, rethinking this issue this morning, I have the following idea:
We should probably just play with the t-cache key but not by adding a new entry in the t-cache list that would be "one per page" as you did initially.
Maybe we could do the following:
- Extract the t-cache attribute value into a `t-set` variable and set that variable into the t-cache instead
- Add a method on website or website.menu which would return a boolean if we detect a slug in one of the menu, like a regex matching "%/word/word-id%"
- Between the t-set of the t-cache variable and the the t-cache instruction, override the t-cache variable to `None` if the method returns `True`

That would actually allow us to re-introduced the `group_ids` on `website.menu`: that method would also return `True` if a `group_ids` is set on a menu to disable the cache.
We had to remove it to be able to introduce t-cache, but it was an useful feature (even if advanced because groups on menu can only be added in debug) for some tech people.

Reply:
I tried what you propose. The problem is how to recognize a record page ? I could do:
1. What you propose is to check that there is a url like that  "%/word/word-id%" but I can create a website.page `/test/gdi-1`
2. A dumb solution would be to hard-code all possible routes (/shop/*, /blog/*,....). => BAD
3. A better solution would be to detect them automatically. Like that:
```py
page_urls = [page.url for page in self.website_id._get_website_pages()]
record_and_pages_urls = [page['loc'] for page in list(self.website_id._enumerate_pages())]
not_record_page_urls = ['/website/info']
record_urls = [url for url in record_and_pages_urls if url not in page_urls and url.count("/") > 1 and url not in not_record_page_urls]
```
It works well but this greatly increases the number of SQL queries required to render a page... In master we could add a compute field to record whether or not the menu contains a URL record. but in stable we can't add field :(

Maybe we should implement the solution 3 in master (with a compute field) and keep the previous solution (do it in JS) for stable. What do you think @rdeodoo ? I pushed the solution 3 (without the new computed field) on the PR (if you want to have a look.

Reply:
> I tried what you propose. The problem is how to recognize a record page ? I could do:
> 
> What you propose is to check that there is a url like that "%/word/word-id%" but I can create a website.page /test/gdi-1

I don't think it's a real issue:
1. People almost never add `/` in their page, even if some people do it. On odoo.com/app/xx we do it but I wonder if it's not because of a retro-compatibility because it was a controller before website.page existed and we don't want to break SEO
2. Even if some do, it's unlikely that they will have the exact format of a slug (/word-id)
3. Even if they do both (multiple `/` and a slug like page name), then it would just wrongly consider it as a controller and disable the cache, which would not be the end of the world imo? At least, it would be the exact (and better) opposite of now: we would lose some perf in this rare corner case, instead of losing perf all the time as now

Unless you disagree, I'd not introduce any code other than the minimum I suggested which should cover 99% of the cases (didn't test)